File: src/14.longest-common-prefix/14.longest-common-prefix.cs
/*
 * @lc app=leetcode id=14 lang=csharp
 *
 * [14] Longest Common Prefix
 */
//  Write a function to find the longest common prefix string amongst an array of strings.

// If there is no common prefix, return an empty string "".

// Example 1:

// Input: ["flower","flow","flight"]
// Output: "fl"
// Example 2:

// Input: ["dog","racecar","car"]
// Output: ""
// Explanation: There is no common prefix among the input strings.
// Note:

// All given inputs are in lowercase letters a-z.
public class Solution
{
    public string LongestCommonPrefix(string[] strs)
    {
        // Accepted
        // 118/118 cases passed (116 ms)
        // Your runtime beats 91.41 % of csharp submissions
        // Your memory usage beats 11.29 % of csharp submissions (23.7 MB)
        string res = "";
        if (strs == null || strs.Length == 0 || strs.Length == 1)
        {
            return strs.Length == 1 ? strs[0] : res;
        }
        for (int i = 0; i < strs[0].Length; i++)
        {
            try
            {
                foreach (string s in strs)
                {
                    if (strs[0][i] != s[i])
                    {
                        return res;
                    }
                }
                res += strs[0][i];
            }
            catch
            {
                return res;
            }
        }
        return res;
    }
}

File: src/14.longest-common-prefix/14.longest-common-prefix.py
#
# @lc app=leetcode id=14 lang=python3
#
# [14] Longest Common Prefix
#
# Write a function to find the longest common prefix string amongst an array of strings.

# If there is no common prefix, return an empty string "".

# Example 1:

# Input: ["flower","flow","flight"]
# Output: "fl"
# Example 2:

# Input: ["dog","racecar","car"]
# Output: ""
# Explanation: There is no common prefix among the input strings.
# Note:

# All given inputs are in lowercase letters a-z.


class Solution:
    def longestCommonPrefix(self, strs: List[str]) -> str:
        # Accepted
        # 118/118 cases passed (36 ms)
        # Your runtime beats 94.02 % of python3 submissions
        # Your memory usage beats 99.68 % of python3 submissions (12.8 MB)
        res = ""
        for i in zip(*strs):
            if len(set(i)) > 1:
                break
            res += i[0]
        return res
